### DISCUSSION TITLE
How to upgrade from separate container version of v1.0.0beta-5

### DISCUSSION THREAD

**Discussion by SteveDinn** [mealie-recipes/mealie#2616] [Q&A] [answered]
I'm currently running Mealie v1.0.0beta-5 in multiple different containers:
- DB (postgres)
- frontend
- API

I've checked out the current [docker-compose.yml](https://github.com/mealie-recipes/mealie/blob/mealie-next/docker/docker-compose.yml) file, and it looks like everything has been combined into one container. What is the new expected format for the docker volume that's mounted under `/app/data/`? The older frontend and API containers seem to share the same data folder, but what happens to the volume that was used by the postgres container?

Are there any other considerations that need to be taken into account before upgrading from v1.0.0beta-5 to v1.0.0RC-1.1?

Thanks!

**Comment by seppeel** [comment, marked as answer]
I started with an pre V1 installation with my migration. So don't delete your old beta5 database and data! Maybe the backup from beta5  is not compatible with the migration tool:

I would go this way:

1. Create a backup in your beta5 installation and save it
2. Stop and remove your beta5 containers
3. Create a new installation as described in https://nightly.mealie.io/documentation/getting-started/installation/sqlite/ or https://nightly.mealie.io/documentation/getting-started/installation/postgres/
4. Adjust alle the settings in the docker compose file. Check this for possible options: https://nightly.mealie.io/documentation/getting-started/installation/backend-config/
5. Start and login to your new installation and go to /group/migrations and import your backup as described in https://nightly.mealie.io/documentation/getting-started/migrating-to-mealie-v1/

**Reply by michael-genson** [reply]
^ this is definitely the recommended approach. For a little bit of context: in the new deployment method, we ditched the frontend container (hence why there is one less container on the docker compose). This is also why you need to expose port 9000 (not 3000, like you used to).

This means for simple deployments (sqlite) you don't even necessarily need docker compose anymore, since it's one container. Since you're using postgres, not much changes for you besides removing the frontend container.

While not necessarily helpful for your case, this might help clarify to anyone else stumbling upon this thread

**Comment by SteveDinn** [comment]
The upgrade went flawlessly. Running 1.0.0RC-1.1 now.

**Comment by threadstone83** [comment]
@SteveDinn can you tell me which docker-image you have choosen? I only have these:
- latest
- api-nightly
- old api-xxx
- dev
- frontend-nightly
- old frontende-xxx
- omni-nightly
- old ones

Is it latest or is it omni-nightly?

**Reply by SteveDinn** [reply]
I'm using "ghcr.io/mealie-recipes/mealie:v1.0.0-RC1.1".

**Reply by threadstone83** [reply]
OK. This is not available at my synology Docker installation. But Thank you!

**Reply by michael-genson** [reply]
If you're using the new container manager on Synology, it supports docker compose which will let you use the ghcr image.

If you're not using the new container manager, you can use SSH (but you should probably upgrade)

**Reply by threadstone83** [reply]
I use the new container manager. Can you explain me to use docker compose? Is this the method were I add the docker-compose-yaml and start it via terminal?

**Reply by michael-genson** [reply]
Under the "Project" tab you can directly enter the docker compose info, then manage it all through the UI (no SSH required):
![image](https://github.com/mealie-recipes/mealie/assets/71845777/b46e59c9-7756-4967-9d88-c0ab0a9c8586)

You'll want to customize your docker compose (such as choosing where you want to mount your filesystem) but here's a basic template to get started:
https://nightly.mealie.io/documentation/getting-started/installation/sqlite/

**Reply by michael-genson** [reply]
The guide uses the nightly image but you can choose whichever (nightly or `ghcr.io/mealie-recipes/mealie:v1.0.0-RC1.1`)

**Reply by threadstone83** [reply]
Perfekt. Thank you very much!
I have added a folder on my synology (volume1/docker/mealie). Did I add it right to the docker-compose.yaml?

`version: "3.7"
services:
  mealie:
    image: ghcr.io/mealie-recipes/mealie:nightly
    container_name: mealie
    ports:
        - "9925:9000" # 
    deploy:
      resources:
        limits:
          memory: 1000M # 
    volumes:
      - /volume1/docker/mealie/data:/app/data/
    environment:
    # Set Backend ENV Variables Here
      - ALLOW_SIGNUP=true
      - PUID=1000
      - PGID=1000
      - TZ=Europe/​Berlin
      - BASE_URL=192.168.178.110
    restart: always

volumes:
  /volume1/docker/mealie/data:
    driver: local`

**Reply by threadstone83** [reply]
I just tried it. The path was wrong. It's not necessary to add the complete path. After changing it, it looks like it works.
Thank you very much michael-genson!!
